onadata/libs/utils/analytics.py

```diff
@@ -135,21 +135,21 @@ def get_event_label(self) -> str:
 
     def get_request_origin(self, request, tracking_properties):
         """Returns the request origin"""
+        event_source = ""  # Initialize event_source variable
         if isinstance(self.tracked_obj, Instance):
+            event_source = "Submission collected from Web"
+            browser_user_agents = ["Chrome", "Mozilla", "Safari"]
+
             try:
                 user_agent = request.META["HTTP_USER_AGENT"]
-                if "Android" in user_agent:
-                    event_source = "Submission collected from ODK COLLECT"
-                elif (
-                    "Chrome" in user_agent
-                    or "Mozilla" in user_agent
-                    or "Safari" in user_agent
-                ):
-                    event_source = "Submission collected from Enketo"
             except KeyError:
-                event_source = ""
-        else:
-            event_source = ""
+                pass
+
+            if "Android" in user_agent:
```

Comment:
I believe `user_agent` won't be defined in scenarios where the KeyError is raised